SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriverDebugContext.cpp

```diff
@@ -70,6 +69,70 @@ void USpatialNetDriverDebugContext::Cleanup()
 	NetDriver->Sender->UpdatePartitionEntityInterestAndPosition();
 }
 
+void USpatialNetDriverDebugContext::AdvanceView()
+{
+	const SpatialGDK::FSubViewDelta& ViewDelta = NetDriver->DebugActorSubView->GetViewDelta();
+	for (const SpatialGDK::EntityDelta& Delta : ViewDelta.EntityDeltas)
+	{
+		switch (Delta.Type)
+		{
+		case SpatialGDK::EntityDelta::ADD:
+			AddComponent(Delta.EntityId);
+			break;
+		case SpatialGDK::EntityDelta::REMOVE:
+			RemoveComponent(Delta.EntityId);
+			break;
+		case SpatialGDK::EntityDelta::TEMPORARILY_REMOVED:
+			RemoveComponent(Delta.EntityId);
+			AddComponent(Delta.EntityId);
+			break;
+		case SpatialGDK::EntityDelta::UPDATE:
+			for (const SpatialGDK::AuthorityChange& Change : Delta.AuthorityLostTemporarily)
+			{
+				if (Change.ComponentSetId == SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID)
+				{
+					AuthorityLost(Delta.EntityId);
+				}
+			}
+			for (const SpatialGDK::AuthorityChange& Change : Delta.AuthorityLost)
+			{
+				if (Change.ComponentSetId == SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID)
+				{
+					AuthorityLost(Delta.EntityId);
+				}
+			}
+			for (const SpatialGDK::ComponentChange& Change : Delta.ComponentUpdates)
+			{
+				if (Change.ComponentId == SpatialConstants::GDK_DEBUG_COMPONENT_ID)
+				{
+					OnComponentChange(Delta.EntityId, Change);
+				}
+			}
+			for (const SpatialGDK::ComponentChange& Change : Delta.ComponentsRefreshed)
+			{
+				if (Change.ComponentId == SpatialConstants::GDK_DEBUG_COMPONENT_ID)
+				{
+					OnComponentChange(Delta.EntityId, Change);
+				}
+			}
+			break;
+		}
+	}
+}
+
+void USpatialNetDriverDebugContext::OnComponentChange(Worker_EntityId EntityId, const SpatialGDK::ComponentChange& Change)
```

Comment:
nit: I think it would make more sense to remove `OnComponentChange` as it merges two paths then forks again in the same way the paths were before. As in, this should probably be `ApplyComponentUpdate` in the foreach update, and the remove/add in for each refresh